### PR TITLE
Use Full-API-Dump.json for typing gen

### PR DESCRIPTION
**Pull request by regginator**
[Info already on Max's repo](https://github.com/MaximumADHD/Roblox-Client-Tracker/pull/27), this is what the typing gen should be using as opposed to the "normal" API dump, which omits many classes and APIs. It's unreliable, and requires redundant 'patches' on class APIs.

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -10,8 +10,8 @@ import { Timer } from "./class/Timer";
 
 const SECURITY_LEVELS = ["None", "PluginSecurity"] as const;
 
-const BASE_URL = "https://raw.githubusercontent.com/MaximumADHD/Roblox-Client-Tracker/roblox/";
-const API_DUMP_URL = BASE_URL + "Mini-API-Dump.json";
+const BASE_URL = "https://github.com/MaximumADHD/Roblox-Client-Tracker/raw/roblox/";
+const API_DUMP_URL = BASE_URL + "Full-API-Dump.json";
```
Comment (Dionysusnu) [inline src/index.ts:14]: Is there a minified version of the fuller API dump available? It's a significant performance difference to download all the unnecessary whitespace.

**Comment by Dionysusnu** [comment]
I can't actually see anything in this version that roblox-ts would _use_ in its types generator. In general, we're not really missing anything from the API dump to generate correct types. What would be the benefit of using this bigger version?

File: src/index.ts [hunk 2]
```diff
@@ -10,8 +10,8 @@ import { Timer } from "./class/Timer";
 
 const SECURITY_LEVELS = ["None", "PluginSecurity"] as const;
 
-const BASE_URL = "https://raw.githubusercontent.com/MaximumADHD/Roblox-Client-Tracker/roblox/";
-const API_DUMP_URL = BASE_URL + "Mini-API-Dump.json";
+const BASE_URL = "https://github.com/MaximumADHD/Roblox-Client-Tracker/raw/roblox/";
```
[the remark this answers is not in the document]
Comment (regginator) [reply src/index.ts:13]: It's not a huge requirement or change, it's just a direct linker from github.com for raw GH UGC returns. (Works with any potential future redirects, etc.)

File: src/index.ts (hunk 1 above)
Comment (regginator) [reply src/index.ts:14]: Not currently, @MaximumADHD only added Full-API-Dump.json as-is for RCT's source.

**Comment by regginator** [comment]
> I can't actually see anything in this version that roblox-ts would _use_ in its types generator. In general, we're not really missing anything from the API dump to generate correct types. What would be the benefit of using this bigger version?

It **guarantees** *all* actual APIs and classes having types on them in TS, without additional patches. It's just a more reliable to work with. Regardless if most of the extras here are Plugin/RobloxScriptSecurity, they *should* be covered if there's a clear ability for them to, as it doesn't make sense to deliberately omit APIs when it's possible to include everything.

**Comment by MaximumADHD** [comment]
I disagree with the notion of this being useful or necessary.
API members added by Full-API-Dump.json can't even be used by scripts, it's only useful in the scope of serialization.

**Comment by Dionysusnu** [comment]
> it doesn't make sense to deliberately omit APIs when it's possible to include everything.

And we aren't. I ran a test with the full API dump and it showed that no changes in output would result from using it. That, combined with Max's explanation of the members being unscriptable, means that there would be no benefit to using this for roblox-ts.
